cloud/aws-functions/orphaned_eks_clusters.py

```diff
@@ -0,0 +1,98 @@
+# Remove expired eks clusters.
+import logging
+import datetime
+import boto3
+import sys
+from time import sleep
+from botocore.exceptions import ClientError
+from boto3.exceptions import Boto3Error
+
+
+def get_clusters_to_terminate(aws_region):
+    clusters_for_deletion = []
+    eks_client = boto3.client('eks', region_name=aws_region)
+    clusters = eks_client.list_clusters()['clusters']
+    if not clusters:
+        logging.info(f"There are no clusters in cloud")
+        sys.exit("There are no clusters in cloud")
+
+    for cluster in clusters:
+        cluster = eks_client.describe_cluster(name=cluster)
+        if 'delete-cluster-after-hours' not in cluster['cluster']['tags'].keys():
+            clusters_for_deletion.append(cluster)
+        else:
+            cluster_lifetime = float(cluster['cluster']['tags']['delete-cluster-after-hours'])
+            current_time = datetime.datetime.now().timestamp()
+            creation_time = datetime.datetime.strptime(str(cluster['cluster']['createdAt']),
+                                                       "%Y-%m-%d %H:%M:%S.%f%z").timestamp()
+            if (current_time - creation_time) / 3600 > cluster_lifetime:
+                clusters_for_deletion.append(cluster)
+
+    if not clusters_for_deletion:
+        logging.info(f"There are no clusters for deletion")
+        sys.exit("There are no clusters for deletion")
+    return clusters_for_deletion
+
+def delete_nodegroup(aws_region, cluster_name):
+    autoscaling_client = boto3.client('autoscaling', region_name=aws_region)
+    ec2 = boto3.resource('ec2')
+
+    autoscaling_group_name = ""
+    for instance in ec2.instances.all():
+        tags = instance.tags
+        tags_dict = {item['Key']: item['Value'] for item in tags}
+        state = instance.state['Name']
+        if tags_dict['alpha.eksctl.io/cluster-name'] and tags_dict[
+            'alpha.eksctl.io/cluster-name'] == cluster_name and state == 'running':
+            autoscaling_group_name = tags_dict['aws:autoscaling:groupName']
+            break
+        else:
+            continue
+
+    try:
+        autoscaling_client.delete_auto_scaling_group(AutoScalingGroupName=autoscaling_group_name, ForceDelete=True)
+    except Boto3Error as e:
+        logging.error(f"Deleting autoscaling group {autoscaling_group_name} failed with error: {e}")
+
+    sleep(200)
```

Comment:
what does this sleep do?

Reply:
we need to wait until autoscaling group will be deleted otherwise we can't delete the cluster. 
It might be better just add the check of deletion. 
What do you think?